### ISSUE TITLE
Implement and qualify an Apple-Silicon training backend for the M3 Ultra

### ISSUE THREAD

**Issue by crates** [entif-ai/bithkuil#64]
## Summary

Make the M3 Ultra a real training accelerator for Bithkuil instead of merely the machine that hosts a CPU-only reference runner.

The current repository pins `torch==2.10.0+cpu`. The reference runner also leaves the student and batches on CPU. #52 already requires local M3 Ultra qualification, but the present implementation cannot measure the machine's Apple GPU training capability.

Implement and qualify an Apple-Silicon training backend that preserves the scientific contract while using Metal effectively.

## Scope

Support one or both of these paths after measured comparison:

- PyTorch MPS/Metal;
- native MLX training.

The first accepted backend is the smallest implementation that can reproduce the reference learner's declared semantics and pass parity tests. Do not maintain two production-quality backends merely for symmetry if one is clearly unsuitable.

The CPU reference remains useful as a deterministic witness and debugging baseline.

## Required parity boundary

The accelerated backend must preserve:

- semantic/token ABI inputs;
- model architecture and parameter accounting;
- initialization profile from #62;
- answer-head semantics;
- optimizer hyperparameters to the extent the backend supports exact parity;
- teacher/curriculum event order;
- checkpoint lineage and resume semantics;
- resource accounting;
- TRAIN/DEV/SEALED separation.

Numerical identity is not required across CPU and GPU kernels. Behavioral parity and declared numerical tolerances are.

Any backend-specific optimizer, precision, kernel, or determinism difference must be recorded rather than silently normalized away.

## M3 Ultra qualification

Measure on the actual local M3 Ultra and record the runtime profile from #52.

At minimum compare:

- CPU reference;
- accelerated backend;
- FP32/BF16 support where scientifically compatible;
- peak unified-memory use;
- examples/sec and tokens/sec;
- optimizer-step latency;
- checkpoint save/load cost;
- end-to-end wall time for the same micro-run;
- numerical divergence in logits/loss/metrics under a fixed seed and fixed example stream.

The goal is not a synthetic kernel trophy. The goal is the fastest local path that still produces trustworthy research artifacts.

## Semantic-initialization support

The accelerated backend must load or compile #62 initialization artifacts without changing their meaning.

Add a pre-step geometry check so backend conversion, dtype casting, or layout transformation cannot silently destroy the intended semantic initialization.

## Ternary boundary

The current W1.58A8 path is float emulation. Do not claim native low-bit acceleration unless a compatible Metal/MLX kernel exists and is measured.

Treat these as separate questions:

```text
Apple GPU training acceleration
!=
native ternary-kernel acceleration
```

#19 remains the owner of ternary training semantics.

## Acceptance criteria

- [ ] At least one Apple-GPU training path runs the complete Bithkuil micro-run on the M3 Ultra.
- [ ] CPU and accelerated paths consume the same frozen semantic/example stream for parity checks.
- [ ] #62 semantic initialization survives backend load/cast within the declared tolerance.
- [ ] Checkpoint/resume works on the accelerated path.
- [ ] Parameter and resource ledgers reconcile.
- [ ] Peak unified memory, throughput and wall time are measured.
- [ ] Backend-specific nondeterminism and numerical differences are documented.
- [ ] CPU reference remains available as an independent witness.
- [ ] No native ternary speed claim is made from float-emulated ternary execution.
- [ ] #52 can use the measured backend evidence to choose the smallest local pilot rung.

## Dependencies

#18 student baseline, #20 checkpoint/recovery, #21 resource accounting, #62 semantic initialization.

## Parent contribution

#52 local M3 Ultra qualification and compute ladder.
Parent roadmap: #5.

## Workability

`EXECUTABLE_LEAF`: one independently verifiable outcome, a qualified Apple-GPU training path with parity and measurement evidence.
